src/main/kotlin/build/buf/gradle/BuildConfiguration.kt

```diff
@@ -28,16 +27,22 @@ private const val BUF_BUILD_PUBLICATION_FILE_BASE_NAME = "image"
 const val BUF_IMAGE_PUBLICATION_NAME = "bufImagePublication"
 
 internal fun Project.configureBuild() {
-    registerBufTask<BuildTask>(BUF_BUILD_TASK_NAME) {
+    registerBufExecTask<BuildTask>(BUF_BUILD_TASK_NAME) {
         group = BUILD_GROUP
         description = "Builds a Buf image from a Protobuf schema."
 
         if (hasProtobufGradlePlugin()) {
             dependsOn(COPY_BUF_CONFIG_TASK_NAME)
-        } else {
-            // Called already during workspace configuration if the protobuf-gradle-plugin has been applied
-            createsOutput()
         }
+
+        inputFiles.setFrom(
+            fileTree(workingDir.get()) {
```

Comment:
This feels a bit spaghetti-ish to me with various `get()` calls distanced from their `set()` calls, but perhaps I just have to adjust to a different syntax style.

Reply:
This is gradle api, I didn't invent it. I don't know it so well, so maybe there is a better way to use it.